tests/potential/test_potential.py

```diff
@@ -308,5 +309,272 @@ def test_fromfile(self):
+
+        #test creation with no standard params, no default values
+        p = self.LinPot(types=('1',), params=('m',))
+        coeff = potential.CoefficientMatrix(types=('1',), params=('m','rmin','rmax','shift'),
+                                            default={'rmin':False,'rmax':False,'shift':False})
+        self.assertCountEqual(p.coeff.types, coeff.types)
+        self.assertCountEqual(p.coeff.params, coeff.params)
+        self.assertDictEqual(p.coeff.default.todict(), coeff.default.todict())
+
+        #test creation with all params, no default values
+        p = self.LinPot(types=('1',), params=('m','rmin','rmax','shift'))
+        coeff = potential.CoefficientMatrix(types=('1',), params=('m','rmin','rmax','shift'),
+                                            default={'rmin':False,'rmax':False,'shift':False})
+        self.assertCountEqual(p.coeff.types, coeff.types)
+        self.assertCountEqual(p.coeff.params, coeff.params)
+        self.assertDictEqual(p.coeff.default.todict(), coeff.default.todict())
+
+    def test_zeros(self):
+        """Test _zeros method"""
+        p = self.LinPot(types=('1',), params=('m',))
+
+        #test with scalar r
+        r = 0.5
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, np.array([r]))
+        np.testing.assert_allclose(u, np.zeros(1))
+        self.assertEqual(s, True)
+
+        u_copy = np.zeros(2)
+        #test with list r
+        r = [0.2, 0.3]
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, r)
+        np.testing.assert_allclose(u, u_copy)
+        self.assertEqual(s, False)
+
+        #test with tuple r
+        r = (0.2, 0.3)
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, r)
+        np.testing.assert_allclose(u, u_copy)
+        self.assertEqual(s, False)
+
+        #test with numpy array r
+        r = np.array([0.2, 0.3])
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, r)
+        np.testing.assert_allclose(u, u_copy)
+        self.assertEqual(s, False)
+
+        #test with non 1-d array r
+        r = np.array([[1, 2], [0.2, 0.3]])
+        with self.assertRaises(TypeError):
+            r_copy, u, s = p._zeros(r)
+
+    def test_energy(self):
+        """Test energy method"""
+        #test with no cutoffs
+        p = self.LinPot(types=('1',), params=('m',), default={'m':2})
+        u = p.energy(pair=('1','1'), r=0.5)
+        self.assertAlmostEqual(u, 1.0)
+        u = p.energy(pair=('1','1'), r=[0.25,0.75])
+        np.testing.assert_allclose(u, [0.5,1.5])
+
+        #test with rmin set
+        p.coeff['1','1']['rmin'] = 0.5
+        u = p.energy(pair=('1','1'), r=0.6)
+        self.assertAlmostEqual(u, 1.2)
+        u = p.energy(pair=('1','1'), r=[0.25,0.75])
+        np.testing.assert_allclose(u, [1.0,1.5])
+
+        #test with rmax set
+        p.coeff['1','1']['rmin'] = False
+        p.coeff['1','1']['rmax'] = 1.5
```

Comment:
Just to mess with the methods as much as possible, let's do this with update:
```suggestion
p.coeff['1','1'].update(rmin=False, rmax=1.5)
```